pwiz_tools/SeeMS/Dialogs/HeatmapForm.cs

```diff
@@ -327,8 +360,24 @@ protected override void OnShown(EventArgs e)
 
                 double scanTime = (double) dgv[scanTimeColumn.Index, i].Value;
                 double ionMobility = (double) dgv[ionMobilityColumn.Index, i].Value;
+                if (ionMobility == 0)

```

Comment:
@bspratt At this point, when ionMobility == 0, I check whether the spectrum has a mobility array. If it does, then I add all the mobility bins from that array. For an MS1, all heatmap data comes from a single spectrum with 3 arrays rather than 1000 spectra with 2 arrays. The number of data points is that same but the performance is much better with 1 spectrum due to data locality.